xdsl/transforms/convert_stencil_to_csl_stencil.py

```diff
@@ -331,14 +313,61 @@ def get_op_split(
                         a.append(use.operation)
                         rem.remove(use.operation)
 
-    if len(a_exports) == 1:
-        return a, b + rem
+    cnst_exports = [cnst for cnst in a_exports if isinstance(cnst, arith.Constant)]
+    if len(a_exports) == 1 + len(cnst_exports):
+        recv_chunk_ops, done_exch_ops = list[Operation](), list[Operation]()
+        for op in ops:
+            if op in a:
+                recv_chunk_ops.append(op)
+                if op in cnst_exports:
+                    done_exch_ops.append(cln := op.clone())
+                    op.result.replace_by_if(
```

Comment:
I'm not sure, but I think this might have to call `rewriter.handle_operation_modification`? Though I have no idea how this works with `replace_by_if`

Reply:
Theoretically possible, practically the op refactoring of the pass is quite substantial, dividing the ops over two regions. This particular `replace_by_if` replaces an arith.const operand with an exact copy of the arith.const operand inserted in the correct region, which may not have to be flagged as a meaningful change.

Reply:
Added some more comments in that region of the code.